CleverFerret/src/main/java/com/universalmedialibrary/services/search/UniversalSearchEngine.kt

```
@@ -0,0 +1,637 @@
+    /**
+     * Execute a saved search
+     */
+    suspend fun executeSavedSearch(savedSearch: SavedSearchEntity): UniversalSearchResults {
+        val mediaTypes = savedSearch.mediaTypes?.let {
+            json.decodeFromString<List<String>>(it)
+        } ?: emptyList()
+
+        val filters = savedSearch.filterCriteria?.let {
+            try {
+                json.decodeFromString<UniversalSearchFilters>(it)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)